test/testdata/boards.local.txt

```diff
@@ -0,0 +1,26 @@
+nessuno.name=Arduino Nessuno
```

Comment:
Hi @silvanocerza. Is this file used anywhere? I couldn't find any references to it.

Reply:
It's used in `PackageManager.loadBoards()`.

https://github.com/arduino/arduino-cli/blob/0ee486d622ba9a73d7d2b507e0383f4750bc6360/arduino/cores/packagemanager/loader.go#L313-L344

Reply:
OK, but that's just the general code for handling boards.local.txt files. I'm asking about this specific `test/testdata/boards.local.txt` file that was added in this PR. Which integration test uses it? I deleted it from this branch in my fork and the integration tests still passed:
https://github.com/per1234/arduino-cli/commit/72d9d0d54d0aa20a66079db77ec2379606968c43

Reply:
It's used by `TestLoadHardware` in `hardware_loader_test.go` but there's no integration tests that verifies this behaviour, I could add it now that I think about it. :thinking: 

By the way if you ran the tests locally it could be that they were still in the `downloaded_hardware/arduino/avr` folder, that's why it might have worked anyway, I should probably delete them after that test finishes or it could pollute other tests.

Reply:
>It's used by TestLoadHardware in hardware_loader_test.go

Here?:
https://github.com/arduino/arduino-cli/blob/384052ab868691d9bf88b9056044ccebdd89e409/legacy/builder/test/hardware_loader_test.go#L32

But, isn't that test using this file?
https://github.com/arduino/arduino-cli/blob/scerza/fix-legacy-platforms-loading/legacy/builder/test/custom_local_txts/boards.local.txt

That's a different file from the one I'm asking about:
https://github.com/arduino/arduino-cli/blob/scerza/fix-legacy-platforms-loading/test/testdata/boards.local.txt
This is in the `/test/testdata/` folder, so I'd expect it to be test data for one of the Python integration test, not one of the Go tests.

Reply:
AHA! I didn't mean to commit the one in `arduino-cli/test/testdata`, it's meant for another PR. I'll remove it, thanks for catching that Per!